scalagen-sbt/src/main/scala/scala/meta/gen/ScalagenPlugin.scala

```diff
@@ -0,0 +1,85 @@
+package scala.meta.gen
+
+import java.io.File
+import java.nio.file.Path
+
+import org.scalameta.scalagen.Runner
+import sbt._
+import sbt.Keys._
+
+import scala.meta._
+
+object ScalagenTags {
+  val SourceGeneration = Tags.Tag("source-generation")
+}
+
+object ScalagenPlugin extends AutoPlugin {
+  override def trigger = allRequirements
+
+  object autoImport {
+
+    lazy val listGenerators =
+      taskKey[Unit]("Lists all generators enabled")
+
+    lazy val generate =
+      taskKey[Seq[File]]("Applies scalagen Generators to sources")
+
+    lazy val generateRecursive =
+      settingKey[Boolean]("Should scalagen recursively generate definitions")
+    lazy val generators = settingKey[Set[Generator]]("Set of Generators to use")
+
+    lazy val baseGenerateSettings: Seq[Def.Setting[_]] = Seq(
```

Comment:
This can be left out of autoImport, autoimport should only contain settings/tasks people are expected to customize in build.sbt